src/agenda/index.ts

```diff
@@ -1,68 +1,115 @@
-    const response = await fetch(`${getApiServer()}/agenda/${uuid}`, {
-        method: 'GET',
-        headers: {
-            'Content-Type': 'application/json',
-        },
-    });
-
-    if (response.status !== 200) {
-        throw new ApiGetError('Unable to get agenda element');
-    }
-
-    return (await response.json()) as AgendaEntry;
-};
-
-export const createAgendaEntry = async (title: string, description: string, event_uuid: string, time: number) => {
-    const response = await fetch(`${getApiServer()}/agenda`, {
-        method: 'PUT',
-        headers: {
-            'Content-Type': 'application/json',
-            'X-Phoenix-Auth': await Oauth.getToken(),
-        },
-        body: JSON.stringify({
-            title,
-            description,
-            event_uuid,
-            time
-        })
-    })
-    return response.status === 200
-}
-
-export const deleteAgendaEntry = async (uuid: string) => {
-    const response = await fetch(`${getApiServer()}/agenda/${uuid}`, {
-        method: 'DELETE',
-        headers: {
-            'Content-Type': 'application/json',
-            'X-Phoenix-Auth': await Oauth.getToken()
-        }
-    })
-    return response.status === 200
+import {getApiServer} from "../meta";
+import * as Oauth from "../user/oauth";
+import {ApiGetError} from "../errors";
+
+interface AgendaEntry {
+    uuid: string;
+    time: number;
+    title: string;
+    description: string;
+}
+
+export const getAgenda = async (): Promise<Array<AgendaEntry>> => {
+    const response = await fetch(`${getApiServer()}/agenda/`, {
+        method: 'GET',
+        headers: {
+            'Content-Type': 'application/json',
+        },
+    });
+
+    if (response.status !== 200) {
+        throw new ApiGetError('Unable to get agenda');
+    }
+
+    return (await response.json()) as Array<AgendaEntry>;
+};
+
+export const getAgendaElement = async (uuid: string): Promise<AgendaEntry> => {
+    const response = await fetch(`${getApiServer()}/agenda/${uuid}`, {
+        method: 'GET',
+        headers: {
+            'Content-Type': 'application/json',
+        },
+    });
+
+    if (response.status !== 200) {
+        throw new ApiGetError('Unable to get agenda element');
+    }
+
+    return (await response.json()) as AgendaEntry;
+};
+
+export const createAgendaEntry = async (
+        event_uuid: string, 
+        title: string, 
+        description: string, 
+        location: string, 
+        time: number, 
+        pinned: boolean
+    ) => {
+    const response = await fetch(`${getApiServer()}/agenda`, {
+        method: 'PUT',
+        headers: {
+            'Content-Type': 'application/json',
+            'X-Phoenix-Auth': await Oauth.getToken(),
+        },
+        body: JSON.stringify({
+            event_uuid,
+            title,
+            description,
+            location,
+            time,
+            pinned
+        })
+    })
+    return response.ok

```

Comment:
the use of `response.ok` is good, but i'd prefer if you could throw a proper exception so we can pass error messages to the user.

You can use this as an example of how to handle an error "properly": https://github.com/phoenixlan/phoenixjs/blob/master/src/ticketVoucher/index.ts#L39

(Replace `Post` in `ApiPostError` with `Put` in this case)